group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1317,6 +1316,9 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
                 throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
             }
+            if (apiVersion >= 2) {
```

Comment:
nit: Let's add a constant to ConsumerGroupHeartbeatRequest for the required version. It will make the code clearer too.

Reply:
Cool. 
I will add and name this constant as `CONSUMER_GENERATED_MEMBER_ID_REQUIRED_VERSION`.